source/extensions/common/wasm/wasm.h

```diff
@@ -183,6 +182,34 @@ void clearCodeCacheForTesting();
 void setTimeOffsetForCodeCacheForTesting(MonotonicTime::duration d);
 WasmEvent toWasmEvent(const std::shared_ptr<WasmHandleBase>& wasm);
 
+class PluginConfig : Logger::Loggable<Logger::Id::wasm> {
+public:
+  PluginConfig(const envoy::extensions::wasm::v3::PluginConfig& config,
+               Server::Configuration::ServerFactoryContext& context, Stats::Scope& scope,
+               Init::Manager& init_manager, envoy::config::core::v3::TrafficDirection direction,
+               const envoy::config::core::v3::Metadata* metadata, bool singleton);
+
+  std::shared_ptr<Context> createContext();
+  Wasm* wasmOfHandle();
```

Comment:
How about making this `WasmHandleSharedPtr wasmHandle()`? That would avoid the possibility of lifetime issues, as well as the name `wasmOfHandle` which sounds a little awkward to me.

Reply:
If we return a `WasmHandleSharedPtr` here, the WasmHandleSharedPtr self may be null. It make the caller need to do additional check to get a Wasm* pointer.

But I agree the naming is not good, may call it wasm() directly?

Reply:
Thanks, I agree `wasm()` is a better name, LG.